### ISSUE TITLE
jsx-no-undef does not support dynamic JSX tags

### ISSUE THREAD

**Issue by darrenscerri** [yannickcr/eslint-plugin-react#484]
```js
class Component extends React.Component {
  render() {
    return (
      <this.props.tag>Test</this.props.tag>
    );
  }
}
```

Eslint is complaining:

`react/jsx-no-undef 'undefined' is not defined`

**Comment by ljharb** [comment]
@darrenscerri Duplicate of #481 and #469. Fixed in ebab180 but not published yet.

**Comment by yannickcr** [comment]
Fixed in `v4.2.0`

**Comment by darrenscerri** [comment]
Issue still persists in `v4.2.0` when using `this`.
